components/PoeModal.tsx

```diff
@@ -0,0 +1,659 @@
+                      );
+
+                      if (
+                        !contract.entrypoints.entrypoints[
+                          detail.parameters.entrypoint
+                        ]
+                      ) {
+                        throw new Error(
+                          `'${detail.parameters.entrypoint}' is not a valid entrypoint for ${detail.destination}`
+                        );
+                      }
+
+                      const methodSchema = new Schema(
+                        contract.entrypoints.entrypoints[
+                          detail.parameters.entrypoint
+                        ]
+                      );
+
+                      const value = contract.methodsObject[
+                        detail.parameters.entrypoint
+                      ](
+                        methodSchema.Execute(detail.parameters.value)
+                      ).toTransferParams().parameter?.value;
+
+                      if (!value) return undefined;
+
+                      const param = emitMicheline(value as Expr);
+
+                      const parser = new Parser();
+
+                      const type = emitMicheline(
+                        parser.parseJSON(
+                          contract.entrypoints.entrypoints[
+                            detail.parameters.entrypoint
+                          ]
+                        ),
+                        {
+                          indent: "",
+                          newline: "",
+                        }
+                      );
+
+                      return {
+                        type: "contract",
+                        values: {
+                          lambda: generateExecuteContractMichelson(version, {
+                            address: detail.destination,
+                            amount: Number(detail.amount),
+                            entrypoint: detail.parameters.entrypoint,
+                            type,
+                            param,
+                          }),
+                          metadata: null,
+                        },
+                      };
+                    } catch (e) {
+                      const errorMessage = `Failed to create a TzSafe proposal: ${
+                        (e as Error).message
+                      }`;
+                      setTransactionError(errorMessage);
+                      console.log("Contract conversion error:", e);
+
+                      state.p2pClient?.sendError(
+                        message.id,
+                        errorMessage,
+                        BeaconErrorType.TRANSACTION_INVALID_ERROR
+                      );
+                      return undefined;
+                    }
+                  } else
+                    return {
+                      type: "transfer",
+                      values: {
+                        to: detail.destination,
+                        amount: detail.amount,
```

Comment:
Need to check if this value is valid.
Sent error back to client for the invalid's case.